typings/vscode-mssql.d.ts

```diff
@@ -489,121 +489,26 @@ declare module 'vscode-mssql' {
 		executeAndScript = 2
 	}
 
-	/**
-	 * Represents boolean value type deployment option.
-	 */
 	export interface DacDeployOptionPropertyBoolean {
 		value: boolean;
 		description: string;
-		displayName: string;
+		propertyName: string;
 	}
 
-	/**
-	 * Represents object type valued deployment option.
-	 */
 	export interface DacDeployOptionPropertyObject {
-		value: SchemaObjectType[];
+		/**
+		 * represents array of number, ex: for Include object types options, the selected options are the enum numbers that value is holding
```

Comment:
This only expect the list of enum values, as this is used for a single property. since with current functionality we are sending an array of numbers to STS. please let me know if this comment is confusing and suggest me the correct way to address it, thanks :)

Reply:
Definitely update the comment to specify exactly what is expected, currently it's very ambiguous about what it contains (and saying it "represents an array of numbers" isn't very helpful. That's literally what it is which is clear enough from the definition itself)

And could you explain in what scenario an array of enum values is used? Mostly for my understanding, just want to make sure that it's actually being used correctly. @kisantia does this stuff make sense to you?

Reply:
Here ExludeObjectItems are coming from an Enum, from STS we are sending some default options to be excluded with ExludeObjectItems parameters, the current functionality is from STS sending the list of enum values to ADS and we use those to uncheck the options on UI by default, when the user modifies the options then we are sending the Enum values of that options back to STS through ExludeObjectItems.

In other PR @kisantia suggested to remove sending the Enum values from STS and send the names instead. I can make those changes and can change the type to string[]. But I feel handling with enum values would be clean and easy! What do you think @Charles-Gagnon @kisantia

Reply:
@llali mentioned something in another PR about the enum values potentially changing order. Is that actually potentially an issue? (that seems dangerous in general so I'm not sure it's actually something we should be doing if we are...there's lots of cases where the enum value is going to be stored or transferred as the underlying number value so changing what that means could break things)

If the number values aren't going to change then I'd generally opt for that - it's a bit more annoying to figure out what the values actually mean (especially on the TS side) but given that C# just sees them all as numbers anyways that's generally going to be easier to work with.

Reply:
Even though the Enum order changes, we are not saving them anywhere and these Enum values are not hardcoded anywhere, as these are directly coming from DacFx API and loading on ADS on each tab instance (opening an options tab).

Reply:
yeah it would be nice to just use the strings so that the TS side only needs to know about the strings and not keep track of both the strings and numbers of an object type, but that can be a cleanup thing later.

Reply:
Sure will make these changes with IncludeObjects PR!